ibis/backends/datafusion/__init__.py

```diff
@@ -628,7 +628,8 @@ def create_table(
                 )
             )
         elif obj is not None:
-            _read_in_memory(obj, name, self, overwrite=overwrite)
+            table_ident = str(sg.table(name, db=database, quoted=quoted))
```

Comment:
```suggestion
            table_ident = sg.table(name, db=database, quoted=quoted).sql(self.dialect)
```

Just to be safe here and make sure we're using the datafusion dialect.